### DISCUSSION TITLE
failed conversion using boost::any_cast

### DISCUSSION THREAD

**Discussion by yzhang6708** [giorgk/ichnos#12] [Q&A]
I created INOCHOS files for a small model. I run the code with version 0.4.13.
In the new version, it didn't recognize increaseRateChange and limitUpperDecreaseStep so I commented them out. Then I got the following error message:
_Ichnos will run using 1 processors
Reading input data...
--> Configuration file: Lakes_config.ini
boost::bad_any_cast: failed conversion using boost::any_cast_

Do you have any thoughts on this? I put my configuration file below:

# Model configuration file for SUTRA 3D Lakes example
# Input file version 0.4.13
[Velocity]
XYZType = CLOUD
Type = DETRM
ConfigFile = Lakes_vf_config.ini

[Domain]
Outline = Lakes_outline.ich
TopFile = Lakes_TopBottom.ich
#TopRadius = 500
#TopPower = 3
BottomFile = 
#BottomRadius = 500
#BottomPower = 0
ProcessorPolys =
#ExpandedPolys = c2vsim_Single_proc_poly.ich

[StepConfig]
Method = RK45
Direction = -1
StepSize = 3600
StepSizeTime = 3600
nSteps = 10
nStepsTime = 0
minExitStepSize = 0.1

[AdaptStep]
MaxStepSize = 10000
MinStepSize = 1000
#increaseRateChange = 1.1
#limitUpperDecreaseStep = 0.15
Tolerance = 10

[StoppingCriteria]
MaxIterationsPerStreamline = 3000
MaxProcessorExchanges = 50
AgeLimit = -1
StuckIter = 10
AttractFile =
AttractRadius = 30

[InputOutput]
ParticleFile = Lakes_particles.ich
#WellFile = Lakes_wells.ich
OutputFile = Lakes_out_
ParticlesInParallel = 5000
GatherOneFile = 0

[Other]
Version = 0.4.13
#Nrealizations = 1

**Comment by giorgk** [comment]
The issue here is that you are using an input file of an earlier version with slightly different options.
Unfortunately I haven't not updated yet the examples for the version 0.4.13. so the input files in the repository are misleading.
However, I have updated the documentation here  https://github.com/giorgk/ichnos/wiki. See the tables with the options and make sure you follow the same spelling and capitalization.

In addition you can run
```
ichnos.exe -h
```
and you can find a list of options of the main input file

I have struggle myself a few times after I change the options. eg. `increaseRateChange`  should now be `IncreaseRateChange` and `limitUpperDecreaseStep ` should be `LimitUpperDecreaseStep` etc

Let me know if there are uncertainties about that

**Reply by yzhang6708** [reply]
Thanks for the reply. I updated the spelling of IncreaseRateChange and limitUpperDecreaseStep. The same error message still popped up:
Ichnos will run using 1 processors
Reading input data...
--> Configuration file: Lakes_config.ini
boost::bad_any_cast: failed conversion using boost::any_cast

**Comment by giorgk** [comment]
This is a strange error. It must be some input inconsistency.
Can you share the console output. I would like to check when this error occurs. 
Can you also try to use Euler or RK2 for a start?

**Reply by yzhang6708** [reply]
I tested Euler and RK. They had the same error message.

`Ichnos will run using 1 processors
Reading input data...
--> Configuration file: Lakes_config.ini
boost::bad_any_cast: failed conversion using boost::any_cast`

**Comment by giorgk** [comment]
In this link you can see an input file that almost match the version you are using. https://raw.githubusercontent.com/wiki/giorgk/ichnos/Mwrkspc/input_files02/c2vsim_CLOUD_SS_config.ini
Can you use this file as template?
Please remove the PECE options and change the method to Euler for start and change the version to 0.4.13.
Let me know how it goes

**Comment by giorgk** [comment]
I believe I found the source of the error.
Do not comment the WellFile. If you don't want to include wells leave it empty.
That's something to fix
